nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutTCP.java

```diff
@@ -0,0 +1,290 @@
+            if (!isConnectionPerFlowFile(context)) {
+                relinquishSender(sender);
+            } else {
+                sender.close();
+            }
+        }
+    }
+
+    /**
+     * event handler method to perform the required actions when a failure has occurred. The FlowFile is penalized, forwarded to the failure relationship and the context is yielded.
+     *
+     * @param context
+     *            - the current process context.
+     *
+     * @param session
+     *            - the current process session.
+     * @param flowFile
+     *            - the FlowFile that has failed to have been processed.
+     */
+    protected void onFailure(final ProcessContext context, final ProcessSession session, final FlowFile flowFile) {
+        session.transfer(session.penalize(flowFile), REL_FAILURE);
+        session.commit();
+        context.yield();
+    }
+
+    /**
+     * Helper method to read the FlowFile content stream into a ByteArrayOutputStream object.
+     *
+     * @param session
+     *            - the current process session.
+     * @param flowFile
+     *            - the FlowFile to read the content from.
+     *
+     * @return ByteArrayOutputStream object containing the FlowFile content.
+     */
+    protected ByteArrayOutputStream readContent(final ProcessSession session, final FlowFile flowFile) {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream((int) flowFile.getSize() + 1);
+        session.read(flowFile, new InputStreamCallback() {
+            @Override
+            public void process(final InputStream in) throws IOException {
+                StreamUtils.copy(in, baos);
+            }
+        });
+
+        return baos;
+    }
+
+    /**
+     * Helper method to append a delimiter to the message contents.
+     *
+     * @param content
+     *            - the message contents.
+     * @param delimiter
+     *            - the delimiter value.
+     *
+     * @return ByteArrayOutputStream object containing the new message contents.
+     */
+    protected ByteArrayOutputStream appendDelimiter(final ByteArrayOutputStream content, final String delimiter, Charset charSet) {
+        content.write(delimiter.getBytes(charSet), 0, delimiter.length());
+        return content;
+    }
+
+    /**
+     * Gets the current value of the "Outgoing Message Delimiter" property.
+     *
+     * @param context
+     *            - the current process context.
+     * @param flowFile
+     *            - the FlowFile being processed.
+     *
+     * @return String containing the Delimiter value.
+     */
+    protected String getDelimiter(final ProcessContext context, final FlowFile flowFile) {
+        String delimiter = context.getProperty(OUTGOING_MESSAGE_DELIMITER).evaluateAttributeExpressions(flowFile).getValue();
+        if (delimiter != null) {
+            delimiter = delimiter.replace("\\n", "\n").replace("\\r", "\r").replace("\\t", "\t");
```

Comment:
My point remains, the characters one enters should be treated as literal values otherwise what may be intuitive to one user is not intuitive to the other especially with NiFi target users consisting of technical and non-technical people. 
As for your comment about translation, If what you say is correct that is a whole different bug and is relatively easy to fix, but IMHO has no relevance on how this particular issue should be handled.